File: leetcode/java/1137.java
class Solution {
  public int tribonacci(int n) {
    if (n == 0)
      return 0;
    if (n == 1)
      return 1;
    if (n == 2)
      return 1;

    var a = 1;
    var b = 1;
    var c = 0;

    for (int i = 2; i < n; i++) {
      var t1 = a;
      var t2 = b;
      a = a + b + c;
      b = t1;
      c = t2;
    }

    return a;
  }
}

File: leetcode/rust/1137.rs
impl Solution {
    pub fn tribonacci(n: i32) -> i32 {
        if n == 2 {
            return 1;
        }
        if n == 1 {
            return 1;
        }
        if n == 0 {
            return 0;
        }

        let mut a = 1;
        let mut b = 1;
        let mut c = 0;

        for _ in 2..n {
            (a, b, c) = (a + b + c, a, b);
        }

        return a;
    }
}
